precompiles/staking/approve_test.go

```diff
@@ -166,24 +170,24 @@ func (s *PrecompileTestSuite) TestApprove() {
 		{
 			"success - remove MsgDelegate authorization",
 			func(contract *vm.Contract) []interface{} {
-				res, err := s.precompile.Approve(s.ctx, s.address, s.stateDB, &method, []interface{}{s.address, big.NewInt(1), []string{staking.DelegateMsg}})
+				res, err := s.precompile.Approve(ctx, s.keyring.GetAddr(0), stDB, &method, []interface{}{s.keyring.GetAddr(0), big.NewInt(1), []string{staking.DelegateMsg}})
 				s.Require().NoError(err)
 				s.Require().Equal(res, cmn.TrueValue)
 
-				authz, expirationTime := s.CheckAuthorization(staking.DelegateAuthz, s.address, s.address)
+				authz, expirationTime := CheckAuthorizationWithContext(ctx, s.network.App.AuthzKeeper, staking.DelegateAuthz, s.keyring.GetAddr(0), s.keyring.GetAddr(0))
```

Comment:
I think it's a bad practice (that was already present in the old tests as well) that we are always creating approvals from the same account to itself. Now that the test suite is so much better we should improve this and instead create the approval for another key.

This would have to be done in all cases :/

```suggestion
				res, err := s.precompile.Approve(ctx, s.keyring.GetAddr(0), stDB, &method, []interface{}{s.keyring.GetAddr(1), big.NewInt(1), []string{staking.DelegateMsg}})
				s.Require().NoError(err)
				s.Require().Equal(res, cmn.TrueValue)

				authz, expirationTime := CheckAuthorizationWithContext(ctx, s.network.App.AuthzKeeper, staking.DelegateAuthz, s.keyring.GetAddr(0), s.keyring.GetAddr(1))
```